packages/cli/src/commands/generate.js

```diff
@@ -66,9 +71,13 @@ export default {
 
     const generator = await cmd.getGenerator(nuxt)
 
-    await generator.generate({
+    const errors = await generator.generate({
       init: true,
       build: cmd.argv.build
     })
+
+    if (cmd.argv['fail-on-page-error'] && errors.errors.length > 0) {
+      throw new Error('Error generating pages, exiting with non-zero code')
+    }
```

Comment:
I was considering to do `throw String` as it is done in run.js - please let me know what's better.

I could also check if `errors` and `errors.errors` is not null, but for the moment from what I see it should always be present. Please let me know if you prefer to add this extra validation.

Reply:
Thanks for the review. Can you please update `generate.generate` to return `{ errors: [] }` ?

Reply:
Sorry, I don;t think I understand. You want to modify `generator.generate` function (it already returns `{errors: []}` if there are no errors)? Or do you want me to return `errors` from `generate.run` instead of throwing an error?

Reply:
I wanted to ensure that generate() function always returns the same data structure like generateResult with `errors` array. So we don't need to check it here.

Reply:
Ah. In fact it does, and the check is not ncecessary. However, I would need to update the mock to behave the same manner. I will do it.

Reply:
Done.